src/proto/generation.proto

```diff
@@ -240,6 +241,92 @@ message ClassifierParameters {
   optional Action realized_action = 3;
 }
 
+//
+// Transforms
+//
+
+enum BorderMode {
+  BORDER_REFLECT = 0;   // reflect image values across the border
+  BORDER_REPLICATE = 1; // replicate border values outside the image
+  BORDER_WRAP = 2;      // wrap around / tile the image values
+  BORDER_ZERO = 3;      // use 0 for locations outside the image
+}
+
+enum ColorMatchMode {
+  COLOR_MATCH_HSV = 0;  // match hue, saturation, and value histograms
+  COLOR_MATCH_LAB = 1;  // match lightness, a, and b histograms
+  COLOR_MATCH_RGB = 2;  // match red, green, and blue histograms
+}
+
+message TransformAddNoise {
+  float amount = 1; // amount of noise to add
+  uint32 seed = 2;  // random seed for noise generation
+}
+
+message TransformBlend {
+  float amount = 1;     // blend amount from 0 to 1
+  Artifact target = 2;  // target image
+}
+
+message TransformColorMatch {
+  ColorMatchMode color_mode = 1;
+  Artifact image = 2;
+}
+
+message TransformContrast {
+  optional float brightness = 1;  // scale on pixel intensities, 1.0 is no change
+  optional float contrast = 2;    // contrast adjustment, 1.0 is no change
+}
+
+message TransformDepthCalc {
+  optional bool export = 1;        // return the depth map as an artifact
+  optional float blend_weight = 2; // blend factor between AdaBins (0.0) and MiDaS (1.0)
+}
+
+message TransformWarp2d {
+  BorderMode border_mode = 1; // extrapolation of border pixels
+  float translate_x = 4;      // horizontal translation in pixels
+  float translate_y = 5;      // vertical translation in pixels
+  float rotate = 2;           // rotation in degrees
+  float scale = 3;            // scale factor
+}
+
+message TransformWarp3d {
+  BorderMode border_mode = 1; // extrapolation of border pixels
+  float translate_x = 3;      // horizontal translation
+  float translate_y = 4;      // vertical translation
+  float translate_z = 5;      // forward translation
+  float rotate_x = 6;         // rotation around X-axis in degrees (pitch)
+  float rotate_y = 7;         // rotation around Y-axis in degrees (yaw)
+  float rotate_z = 8;         // rotation around Z-axis in degrees (roll)
+  float near_plane = 9;       // near clipping plane of view frustum
```

Comment:
What units is this?

Reply:
This has not been addressed.

Reply:
some cursory googling suggests opencv's camera matrix is in units of mm's, so that's my guess for the units used for near_plane and far_plane. FOV is degrees. the 2D translations are unique because there's no camera estimation and are consequently in units of pixels I believe. rotation angles I'm pretty sure are radians. 

if we change fov to radians it'd at least be consistent, but we'd need to translate it back to degrees before passing to pytorch3d. i'm not sure if there's a good solution to put the 2D transforms into something resembling the same scale as the 3D transforms